---
title: "Timecouse_integration"
output: html_document
---


**Session information**
> sessionInfo()
R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C            LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C        LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C         LC_TELEPHONE=C      
[11] LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] car_3.1-1                   carData_3.0-5               caret_6.0-93                lattice_0.20-45             xgboost_1.7.3.1             scPred_1.9.2                SeuratDisk_0.0.0.9020       slingshot_2.5.2            
 [9] TrajectoryUtils_1.4.0       SingleCellExperiment_1.18.1 SummarizedExperiment_1.26.1 Biobase_2.56.0              GenomicRanges_1.48.0        GenomeInfoDb_1.32.4         IRanges_2.30.1              S4Vectors_0.34.0           
[17] BiocGenerics_0.42.0         MatrixGenerics_1.8.1        matrixStats_0.63.0          princurve_2.1.6             RColorBrewer_1.1-3          ggrepel_0.9.3               data.table_1.14.8           elementalist_0.0.0.9000    
[25] patchwork_1.1.2             cowplot_1.1.1               sctransform_0.3.5           SeuratObject_4.1.3          Seurat_4.3.0                forcats_0.5.1               stringr_1.5.0               dplyr_1.1.0                
[33] purrr_1.0.1                 readr_2.1.2                 tidyr_1.3.0                 tibble_3.1.8                ggplot2_3.4.1               tidyverse_1.3.1            

loaded via a namespace (and not attached):
  [1] utf8_1.2.3             spatstat.explore_3.0-6 reticulate_1.28        R.utils_2.11.0         tidyselect_1.2.0       htmlwidgets_1.6.1      grid_4.2.0             Rtsne_0.16             pROC_1.18.0           
 [10] munsell_0.5.0          codetools_0.2-18       ica_1.0-3              future_1.31.0          miniUI_0.1.1.1         withr_2.5.0            spatstat.random_3.1-3  colorspace_2.1-0       progressr_0.13.0      
 [19] knitr_1.42             rstudioapi_0.13        ROCR_1.0-11            tensor_1.5             listenv_0.9.0          labeling_0.4.2         harmony_0.1.1          GenomeInfoDbData_1.2.8 polyclip_1.10-4       
 [28] bit64_4.0.5            farver_2.1.1           parallelly_1.34.0      vctrs_0.5.2            generics_0.1.3         ipred_0.9-13           xfun_0.37              timechange_0.2.0       R6_2.5.1              
 [37] ggbeeswarm_0.7.1       hdf5r_1.3.8            bitops_1.0-7           spatstat.utils_3.0-1   DelayedArray_0.22.0    assertthat_0.2.1       promises_1.2.0.1       scales_1.2.1           nnet_7.3-17           
 [46] beeswarm_0.4.0         gtable_0.3.1           globals_0.16.2         goftest_1.2-3          timeDate_4022.108      rlang_1.0.6            splines_4.2.0          lazyeval_0.2.2         ModelMetrics_1.2.2.2  
 [55] spatstat.geom_3.0-6    broom_0.8.0            yaml_2.3.7             reshape2_1.4.4         abind_1.4-5            modelr_0.1.8           backports_1.4.1        httpuv_1.6.9           lava_1.7.2.1          
 [64] tools_4.2.0            ellipsis_0.3.2         ggridges_0.5.4         Rcpp_1.0.10            plyr_1.8.8             zlibbioc_1.42.0        RCurl_1.98-1.10        rpart_4.1.16           deldir_1.0-6          
 [73] pbapply_1.7-0          zoo_1.8-11             haven_2.5.0            cluster_2.1.3          fs_1.6.1               magrittr_2.0.3         scattermore_0.8        lmtest_0.9-40          reprex_2.0.1          
 [82] RANN_2.6.1             fitdistrplus_1.1-8     hms_1.1.1              mime_0.12              evaluate_0.20          xtable_1.8-4           readxl_1.4.0           gridExtra_2.3          compiler_4.2.0        
 [91] KernSmooth_2.23-20     crayon_1.5.2           R.oo_1.24.0            htmltools_0.5.4        later_1.3.0            tzdb_0.3.0             lubridate_1.9.2        DBI_1.1.2              dbplyr_2.1.1          
[100] MASS_7.3-56            Matrix_1.5-3           cli_3.6.0              R.methodsS3_1.8.1      gower_1.0.1            parallel_4.2.0         igraph_1.4.1           pkgconfig_2.0.3        sp_1.6-0              
[109] plotly_4.10.1          spatstat.sparse_3.0-0  recipes_1.0.5          xml2_1.3.3             foreach_1.5.2          vipor_0.4.5            hardhat_1.2.0          XVector_0.36.0         prodlim_2019.11.13    
[118] rvest_1.0.2            digest_0.6.31          RcppAnnoy_0.0.20       spatstat.data_3.0-0    rmarkdown_2.20         cellranger_1.1.0       leiden_0.4.3           uwot_0.1.14            kernlab_0.9-32        
[127] shiny_1.7.4            lifecycle_1.0.3        nlme_3.1-157           jsonlite_1.8.4         viridisLite_0.4.1      limma_3.52.4           fansi_1.0.4            pillar_1.8.1           ggrastr_1.0.1         
[136] fastmap_1.1.1          httr_1.4.5             survival_3.3-1         glue_1.6.2             png_0.1-8              iterators_1.0.14       bit_4.0.5              class_7.3-20           stringi_1.7.12        
[145] irlba_2.3.5.1          future.apply_1.10.0   



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r packages, eval=TRUE, echo=TRUE}
# loading package(s)
suppressPackageStartupMessages({
library(tidyverse)                                        # for general data wrangling, also includes Seurat dependency dplyr
library(Seurat)                                           # for scRNA-Seq analysis
library(sctransform)                                      # for normalization, feature identification, and scaling
library(cowplot)                                          # add-on to ggplot
library(patchwork)                                        # for plotting
library(elementalist)                                     # ggplot themes
})

# increase globals for sctransform
options(future.globals.maxSize= 1024*1024^3)
```


```{r read, eval=TRUE, echo=TRUE}
# integrate datasets


# read in cardiac eb raw data
eb_raw_data <- Read10X("/YG-EM-ln1-EM314_human/outs/filtered_feature_bc_matrix/")


# create seurat object for cardiac eb
eb_obj <- CreateSeuratObject(counts = eb_raw_data)


# setting directory for deconvolution files
dir <- "/YG-EM-ln1-EM314_human/"

# add vireo information (demultiplex to identify individuals)
add_vireo <- function(eb_obj, dir) {
  donor_ids <- read.table(paste0(dir, 'vireo/donor_ids.tsv'), header = TRUE, stringsAsFactors = FALSE)
  
  eb_obj <- AddMetaData(eb_obj, donor_ids$donor_id,   col.name ='vireo.individual')
  eb_obj <- AddMetaData(eb_obj, donor_ids$prob_max,   col.name ='vireo.prob.singlet')
  
  ambient.file <- paste0(dir, 'vireo/prop_ambient.tsv')
  if (file.exists(ambient.file)) {
    prop <- read.table(paste0(dir, 'vireo/prop_ambient.tsv'), header = TRUE, stringsAsFactors = FALSE)
    
    inds <- unique(donor_ids$best_singlet)
    donor_ids$prop_donor <- 0
    for (i in 1:length(inds)) {
      w <- which(donor_ids$best_singlet==inds[i])
      donor_ids$prop_donor[w] <- prop[w,inds[i]]
    }
    eb_obj <- AddMetaData(eb_obj, donor_ids$prop_donor, col.name ='vireo.prop.donor')
  }
  eb_obj
}

eb_obj <- add_vireo(eb_obj, dir)

individual.colname <- ('vireo.individual')
individual.colname <- individual.colname[which(individual.colname %in% colnames(eb_obj@meta.data))][1]


# filter out doublets and unassigned cells
eb_obj <- subset(x = eb_obj, subset = vireo.individual == "doublet", invert = TRUE)
eb_obj <- subset(x = eb_obj, subset = vireo.individual == "unassigned", invert = TRUE)

# filter out cells with less than 1500 genes
eb_obj <- subset(eb_obj, subset = nFeature_RNA >= 1500)


# read in ipsc timecourse data
ips_obj <- readRDS(file = "Elorbany_2022.rds")

# create timecourse seurat object and filter for features expressed in at least 10 cells and cells with at least 300 features
ips_obj <- CreateSeuratObject(counts = ips_obj@assays$RNA@counts, min.cells = 10, min.features = 300) 

# filter out cells with more than 25% mtDNA
ips_obj[["percent.mt"]] <- PercentageFeatureSet(ips_obj, pattern = "^MT-")
ips_obj <- subset(ips_obj, subset = percent.mt <= 25)

# filter out cells with doublet probability greater than 0.3
ips_obj <- subset(ips_obj, subset = PRB.DBL <= 0.3)

# remove all ambiguous cells
ips_obj <- subset(x = ips_obj, subset = BEST == "AMB", invert = TRUE)

# removing cells not designated as singlets (i.e. individual is n/a)
ips_obj <- subset(ips_obj, subset = Individual != "NA")

# filter cells that have features or reads above or below 4 s.d. from the median
count.max <- round(median(ips_obj$nCount_RNA) + 4 * sd(ips_obj$nCount_RNA), digits = -2)
count.min <- round(median(ips_obj$nCount_RNA) - 4 * sd(ips_obj$nCount_RNA), digits = -2)
feat.max <- round(median(ips_obj$nFeature_RNA) + 4 * sd(ips_obj$nFeature_RNA), digits = -2)
feat.min <- round(median(ips_obj$nFeature_RNA) - 4 * sd(ips_obj$nFeature_RNA), digits = -2)

if (count.min < 0){
   count.min <- 0
} else {
   count.min <- count.min
}
  
if (feat.min < 0){
   feat.min <- 0
} else {
   feat.min <- feat.min
}

ips_obj <- subset(ips_obj, subset = nFeature_RNA > feat.min & nFeature_RNA < feat.max & nCount_RNA < count.max & nCount_RNA > count.min)


# combine objects and and format 

# add origin metadata
ips_obj <- AddMetaData(ips_obj, metadata="ipsc_cms", col.name="Origin")
eb_obj <- AddMetaData(eb_obj, metadata="cardiac_eb", col.name="Origin")

# make list of two objects
combined_obj <- merge(ips_obj, y = eb_obj, add.cell.ids = c("IPSC", "EB"), project = "compare")

# the cardiac EB has no "day" that it was harvested, replacing with EB
combined_obj@meta.data$Day <- combined_obj@meta.data$Day %>% replace_na('EB')

combined_obj <- SplitObject(combined_obj, split.by = "Origin")
combined_obj <- lapply(X = combined_obj, FUN = SCTransform, variable.features.n = 5000)
features <- SelectIntegrationFeatures(object.list = combined_obj, nfeatures = 5000)
combined_obj <- PrepSCTIntegration(object.list = combined_obj, anchor.features = features)
anchors <- FindIntegrationAnchors(object.list = combined_obj, normalization.method = "SCT", anchor.features = features)
combined.sct <- IntegrateData(anchorset = anchors, normalization.method = "SCT")
DefaultAssay(combined.sct) <- "integrated"
combined.sct <- RunPCA(combined.sct, npcs = 50)
combined.sct <- RunUMAP(combined.sct, dims = 1:50)
combined.sct <- FindNeighbors(combined.sct, dims = 1:50)

#change order of day-levels so they are listed chronologically
combined.sct$Day <- factor(combined_obj$Day, levels = c('Day 0', 'Day 1', 'Day 3', 'Day 5', 'Day 7', 'Day 11', 'Day 15', 'EB'))

obj <- combined.sct
```



```{r, eval=TRUE, echo=TRUE}
# set cluster resolution
obj <- FindClusters(obj, resolution = 0.1)
DimPlot(obj, reduction = "umap")
DimPlot(obj, group.by = 'Origin', cols = c("orange", "light blue"), order = TRUE, raster = FALSE)
DimPlot(obj, split.by = "Origin")
```


```{r, eval=TRUE, echo=TRUE}
# prepare color palette  
my_seven_colors <- c("#117733", "#882255", "#CC6677", "#999933", "#44AA99", "#332288", "#AA4499")
pie(rep(1, 7), col = my_seven_colors)
```


```{r umap, eval=TRUE, echo=TRUE}
# plot clusters of the integrated/combined object 
all <- DimPlot(obj, reduction = "umap", cols = my_seven_colors, order = FALSE, label = FALSE, raster = FALSE) + ggtitle("UMAP of integrated dataset") + theme(plot.title = element_text(size=11), axis.title.x = element_text(size = 8),  axis.title.y = element_text(size = 8), axis.text.x = element_text(size = 7), axis.text.y = element_text(size = 7))
all
```



```{r day0, eval=TRUE, echo=TRUE}
# highlight ipsc-cms day 0 clusters
cells0_0 <- Cells(subset(x = obj, subset = Day == "Day 0" & seurat_clusters == 0))
cells0_1 <- Cells(subset(x = obj, subset = Day == "Day 0" & seurat_clusters == 1))
cells0_2 <- Cells(subset(x = obj, subset = Day == "Day 0" & seurat_clusters == 2))
cells0_3 <- Cells(subset(x = obj, subset = Day == "Day 0" & seurat_clusters == 3))
cells0_4 <- Cells(subset(x = obj, subset = Day == "Day 0" & seurat_clusters == 4))
cells0_5 <- Cells(subset(x = obj, subset = Day == "Day 0" & seurat_clusters == 5))
#cells0_6 <- Cells(subset(x = obj, subset = Day == "Day 0" & seurat_clusters == 6)) # no cells

day0 <- DimPlot(object = obj, cells.highlight = list(cells0_0, cells0_1, cells0_2, cells0_3, cells0_4, cells0_5), cols.highlight = c("#332288", "#44AA99", "#999933", "#CC6677", "#882255", "#117733"), cols = "#DDDDDD", order = TRUE, label = FALSE, raster = TRUE, raster.dpi = c(1024, 1024)) + ggtitle("Day 0") + theme(plot.title = element_text(hjust = 0.5, size = 11), panel.border = element_rect_round(radius = unit(0.1, "snpc"), color = "black", fill=NA)) + NoLegend() + NoAxes()
day0
```



```{r day1, eval=TRUE, echo=TRUE}
# highlight ipsc-cms day 1 clusters
cells1_0 <- Cells(subset(x = obj, subset = Day == "Day 1" & seurat_clusters == 0))
cells1_1 <- Cells(subset(x = obj, subset = Day == "Day 1" & seurat_clusters == 1))
cells1_2 <- Cells(subset(x = obj, subset = Day == "Day 1" & seurat_clusters == 2))
cells1_3 <- Cells(subset(x = obj, subset = Day == "Day 1" & seurat_clusters == 3))
cells1_4 <- Cells(subset(x = obj, subset = Day == "Day 1" & seurat_clusters == 4))
cells1_5 <- Cells(subset(x = obj, subset = Day == "Day 1" & seurat_clusters == 5))
#cells1_6 <- Cells(subset(x = obj, subset = Day == "Day 1" & seurat_clusters == 6)) # no cells

day1 <- DimPlot(object = obj, cells.highlight = list(cells1_0, cells1_1, cells1_2, cells1_3, cells1_4, cells1_5), cols.highlight = c("#332288", "#44AA99", "#999933", "#CC6677", "#882255", "#117733"), cols = "#DDDDDD", order = TRUE, label = FALSE, raster = TRUE, raster.dpi = c(1024, 1024)) + ggtitle("Day 1") + theme(plot.title = element_text(hjust = 0.5, size = 11), panel.border = element_rect_round(radius = unit(0.1, "snpc"), color = "black", fill=NA)) + NoLegend() + NoAxes()
day1
```


```{r day3, eval=TRUE, echo=TRUE}
# highlight ipsc-cms day 3 clusters
cells3_0 <- Cells(subset(x = obj, subset = Day == "Day 3" & seurat_clusters == 0))
cells3_1 <- Cells(subset(x = obj, subset = Day == "Day 3" & seurat_clusters == 1))
cells3_2 <- Cells(subset(x = obj, subset = Day == "Day 3" & seurat_clusters == 2))
cells3_3 <- Cells(subset(x = obj, subset = Day == "Day 3" & seurat_clusters == 3))
cells3_4 <- Cells(subset(x = obj, subset = Day == "Day 3" & seurat_clusters == 4))
cells3_5 <- Cells(subset(x = obj, subset = Day == "Day 3" & seurat_clusters == 5))
#cells3_6 <- Cells(subset(x = obj, subset = Day == "Day 3" & seurat_clusters == 6)) # no cells

day3 <- DimPlot(object = obj, cells.highlight = list(cells3_0, cells3_1, cells3_2, cells3_3, cells3_4, cells3_5), cols.highlight = c("#332288", "#44AA99", "#999933", "#CC6677", "#882255", "#117733"), cols = "#DDDDDD", order = TRUE, label = FALSE, raster = TRUE, raster.dpi = c(1024, 1024)) + ggtitle("Day 3") + theme(plot.title = element_text(hjust = 0.5, size = 11), panel.border = element_rect_round(radius = unit(0.1, "snpc"), color = "black", fill=NA)) + NoLegend() + NoAxes()
day3
```



```{r day5, eval=TRUE, echo=TRUE}
# highlight ipsc-cms day 5 clusters
cells5_0 <- Cells(subset(x = obj, subset = Day == "Day 5" & seurat_clusters == 0))
cells5_1 <- Cells(subset(x = obj, subset = Day == "Day 5" & seurat_clusters == 1))
cells5_2 <- Cells(subset(x = obj, subset = Day == "Day 5" & seurat_clusters == 2))
cells5_3 <- Cells(subset(x = obj, subset = Day == "Day 5" & seurat_clusters == 3))
cells5_4 <- Cells(subset(x = obj, subset = Day == "Day 5" & seurat_clusters == 4))
cells5_5 <- Cells(subset(x = obj, subset = Day == "Day 5" & seurat_clusters == 5))
cells5_6 <- Cells(subset(x = obj, subset = Day == "Day 5" & seurat_clusters == 6))

day5 <- DimPlot(object = obj, cells.highlight = list(cells5_0, cells5_1, cells5_2, cells5_3, cells5_4, cells5_5, cells5_6), cols.highlight = c("#AA4499", "#332288", "#44AA99", "#999933", "#CC6677", "#882255", "#117733"), cols = "#DDDDDD", order = TRUE, label = FALSE, raster = TRUE, raster.dpi = c(1024, 1024)) + ggtitle("Day 5") + theme(plot.title = element_text(hjust = 0.5, size = 11), panel.border = element_rect_round(radius = unit(0.1, "snpc"), color = "black", fill=NA)) + NoLegend() + NoAxes()
day5
```



```{r day7, eval=TRUE, echo=TRUE}
# highlight ipsc-cms day 7 clusters
cells7_0 <- Cells(subset(x = obj, subset = Day == "Day 7" & seurat_clusters == 0))
cells7_1 <- Cells(subset(x = obj, subset = Day == "Day 7" & seurat_clusters == 1))
cells7_2 <- Cells(subset(x = obj, subset = Day == "Day 7" & seurat_clusters == 2))
cells7_3 <- Cells(subset(x = obj, subset = Day == "Day 7" & seurat_clusters == 3))
cells7_4 <- Cells(subset(x = obj, subset = Day == "Day 7" & seurat_clusters == 4))
cells7_5 <- Cells(subset(x = obj, subset = Day == "Day 7" & seurat_clusters == 5))
cells7_6 <- Cells(subset(x = obj, subset = Day == "Day 7" & seurat_clusters == 6))

day7 <- DimPlot(object = obj, cells.highlight = list(cells7_0, cells7_1, cells7_2, cells7_3, cells7_4, cells7_5, cells7_6), cols.highlight = c("#AA4499", "#332288", "#44AA99", "#999933", "#CC6677", "#882255", "#117733"), cols = "#DDDDDD", order = TRUE, label = FALSE, raster = TRUE, raster.dpi = c(1024, 1024)) + ggtitle("Day 7") + theme(plot.title = element_text(hjust = 0.5, size = 11), panel.border = element_rect_round(radius = unit(0.1, "snpc"), color = "black", fill=NA)) + NoLegend() + NoAxes()
day7
```



```{r day11, eval=TRUE, echo=TRUE}
# highlight ipsc-cms day 11 clusters
cells11_0 <- Cells(subset(x = obj, subset = Day == "Day 11" & seurat_clusters == 0))
cells11_1 <- Cells(subset(x = obj, subset = Day == "Day 11" & seurat_clusters == 1))
cells11_2 <- Cells(subset(x = obj, subset = Day == "Day 11" & seurat_clusters == 2))
cells11_3 <- Cells(subset(x = obj, subset = Day == "Day 11" & seurat_clusters == 3))
cells11_4 <- Cells(subset(x = obj, subset = Day == "Day 11" & seurat_clusters == 4))
cells11_5 <- Cells(subset(x = obj, subset = Day == "Day 11" & seurat_clusters == 5))
cells11_6 <- Cells(subset(x = obj, subset = Day == "Day 11" & seurat_clusters == 6))

day11 <- DimPlot(object = obj, cells.highlight = list(cells11_0, cells11_1, cells11_2, cells11_3, cells11_4, cells11_5, cells11_6), cols.highlight = c("#AA4499", "#332288", "#44AA99", "#999933", "#CC6677", "#882255", "#117733"), cols = "#DDDDDD", order = TRUE, label = FALSE, raster = TRUE, raster.dpi = c(1024, 1024)) + ggtitle("Day 11") + theme(plot.title = element_text(hjust = 0.5, size = 11), panel.border = element_rect_round(radius = unit(0.1, "snpc"), color = "black", fill=NA)) + NoLegend() + NoAxes()
day11
```


```{r day15, eval=TRUE, echo=TRUE}
# highlight ipsc-cms day 15 clusters
cells15_0 <- Cells(subset(x = obj, subset = Day == "Day 15" & seurat_clusters == 0))
cells15_1 <- Cells(subset(x = obj, subset = Day == "Day 15" & seurat_clusters == 1))
cells15_2 <- Cells(subset(x = obj, subset = Day == "Day 15" & seurat_clusters == 2))
cells15_3 <- Cells(subset(x = obj, subset = Day == "Day 15" & seurat_clusters == 3))
cells15_4 <- Cells(subset(x = obj, subset = Day == "Day 15" & seurat_clusters == 4))
cells15_5 <- Cells(subset(x = obj, subset = Day == "Day 15" & seurat_clusters == 5))
cells15_6 <- Cells(subset(x = obj, subset = Day == "Day 15" & seurat_clusters == 6))

day15 <- DimPlot(object = obj, cells.highlight = list(cells15_1, cells15_2, cells15_3, cells15_4, cells15_5, cells15_6), cols.highlight = c("#AA4499", "#332288", "#44AA99", "#999933", "#CC6677", "#882255", "#117733"), cols = "#DDDDDD", order = TRUE, label = FALSE, raster = TRUE, raster.dpi = c(1024, 1024)) + ggtitle("Day 15") + theme(plot.title = element_text(hjust = 0.5, size = 11), panel.border = element_rect_round(radius = unit(0.1, "snpc"), color = "black", fill=NA)) + NoLegend() + NoAxes()
day15
```



```{r EB, eval=TRUE, echo=TRUE}
# highlight cardiac EB clusters
cellsEB_0 <- Cells(subset(x = obj, subset = Day == "EB" & seurat_clusters == 0))
cellsEB_1 <- Cells(subset(x = obj, subset = Day == "EB" & seurat_clusters == 1))
cellsEB_2 <- Cells(subset(x = obj, subset = Day == "EB" & seurat_clusters == 2))
cellsEB_3 <- Cells(subset(x = obj, subset = Day == "EB" & seurat_clusters == 3))
cellsEB_4 <- Cells(subset(x = obj, subset = Day == "EB" & seurat_clusters == 4))
cellsEB_5 <- Cells(subset(x = obj, subset = Day == "EB" & seurat_clusters == 5))
cellsEB_6 <- Cells(subset(x = obj, subset = Day == "EB" & seurat_clusters == 6))
EB <- DimPlot(object = obj, cells.highlight = list(cellsEB_0, cellsEB_1, cellsEB_2, cellsEB_3, cellsEB_4, cellsEB_5, cellsEB_6), cols.highlight = c("#AA4499", "#332288", "#44AA99", "#999933", "#CC6677", "#882255", "#117733"), cols = "#DDDDDD", order = TRUE, label = FALSE, raster = TRUE, raster.dpi = c(1024, 1024)) + ggtitle("Guided differentiation") + theme(plot.title = element_text(hjust = 0.5, size = 11), panel.border = element_rect_round(radius = unit(0.1, "snpc"), color = "black", fill=NA)) + NoLegend() + NoAxes()
EB
```



```{r umaps, eval=TRUE, echo=TRUE}
# setup layout for printing multiple plots
umap_layout <- c(area(1, 1), area(1, 2), area(1, 3), area(2, 1), area(2, 2), area(2, 3), area(3, 1), area(3, 2, 4, 3))

# plot each ipsc-cm day and the cardiac EB
day0 + day1 + day3 + day5 + day7 + day11 + day15 + EB + plot_layout(design = umap_layout)
```




